akka-bench-jmh/src/main/scala/akka/actor/TellOnlyBenchmark.scala

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.actor
+
+import akka.dispatch._
+import akka.testkit.TestProbe
+import akka.util.Helpers.ConfigOps
+import com.typesafe.config.{ Config, ConfigFactory }
+import java.util.concurrent.TimeUnit
+import org.openjdk.jmh.annotations._
+import scala.concurrent.duration._
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.SingleShotTime))
+@Fork(5)
+@Threads(1)
+@Warmup(iterations = 10, batchSize = TellOnlyBenchmark.numMessages)
+@Measurement(iterations = 10, batchSize = TellOnlyBenchmark.numMessages)
+class TellOnlyBenchmark {
+  import TellOnlyBenchmark._
+
+  implicit var system: ActorSystem = _
+
+  @Setup(Level.Trial)
+  def setup() {
+    system = ActorSystem("TellOnlyBenchmark", ConfigFactory.parseString(
+      s"""| akka {
+          |   log-dead-letters = off
+          |   actor {
+          |     default-dispatcher {
+          |       executor = "fork-join-executor"
+          |       fork-join-executor {
+          |         parallelism-min = 1
+          |         parallelism-max = 4
+          |       }
+          |       throughput = 1
+          |     }
+          |   }
+          | }
+          | dropping-dispatcher {
+          |   fork-join-executor.parallelism-min = 1
+          |   fork-join-executor.parallelism-max = 1
+          |   type = "akka.actor.TellOnlyBenchmark$$DroppingDispatcherConfigurator"
+          |   mailbox-type = "akka.actor.TellOnlyBenchmark$$UnboundedDroppingMailbox"
+          | }
+          | """.stripMargin))
+  }
+
+  @TearDown(Level.Trial)
+  def shutdown() {
+    system.shutdown()
+    system.awaitTermination()
+  }
+
+  var actor: ActorRef = _
+  var probe: TestProbe = _
+
+  @Setup(Level.Iteration)
+  def setupIteration() {
+    actor = system.actorOf(Props[TellOnlyBenchmark.Echo].withDispatcher("dropping-dispatcher"))
+    probe = TestProbe()
+    probe.watch(actor)
+    probe.send(actor, message)
+    probe.expectMsg(message)
+    probe.send(actor, flipDrop)
+    probe.expectNoMsg(200.millis)
+    System.gc()
+  }
+
+  @TearDown(Level.Iteration)
+  def shutdownIteration() {
+    probe.send(actor, flipDrop)
+    probe.expectNoMsg(200.millis)
+    actor ! stop
+    probe.expectTerminated(actor, timeout)
+    actor = null
+    probe = null
+  }
+
+  @Benchmark
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  def tell() = {
+    probe.send(actor, message)
+  }
+}
+
+object TellOnlyBenchmark {
+  final val stop = "stop"
+  final val message = "message"
+  final val flipDrop = "flipDrop"
+  final val timeout = 5.seconds
+  final val numMessages = 1000000
+
+  class Echo extends Actor {
+    def receive = {
+      case s @ `stop` ⇒
+        context stop self
+      case m ⇒ sender ! m
+    }
+  }
+
+  class DroppingMessageQueue extends UnboundedMailbox.MessageQueue {
+    @volatile var dropping = false
```

Comment:
I want to measure the cost of the `tell` code without the mailbox implementation, and I also don't want the other actor to wake up and skew the measurement.